test/IISIntegration.FunctionalTests/Inprocess/SynchronousReadAndWriteTests.cs

```diff
@@ -70,55 +67,73 @@ public SynchronousReadAndWriteTests(IISTestSiteFixture fixture)
-                    "\r\n";
-                var bytes = 0;
-                var requestStringBytes = Encoding.ASCII.GetBytes(request);
-                var testStringBytes = Encoding.ASCII.GetBytes(testString);
+                    "\r\n" + testString;
+
+                foreach (var c in request)
+                {
+                    await connection.Send(c.ToString());
+                    await Task.Delay(10);
+                }
+
+                await connection.Receive(
+                    "HTTP/1.1 200 OK",
+                    "");
+                await connection.ReceiveHeaders();
 
-                while ((bytes += socket.Send(requestStringBytes, bytes, 1, SocketFlags.None)) < requestStringBytes.Length)
+                for (int i = 0; i < 100; i++)
                 {
+                    foreach (var c in testString)
+                    {
+                        await connection.Receive(c.ToString());
+                    }
+                    await Task.Delay(10);
                 }
+                await connection.WaitForConnectionClose();
+            }
+        }
 
-                bytes = 0;
-                while ((bytes += socket.Send(testStringBytes, bytes, 1, SocketFlags.None)) < testStringBytes.Length)
+        [ConditionalFact]
+        public async Task ReadAndWriteInterleaved()
+        {
+            using (var connection = _fixture.CreateTestConnection())
+            {
+                var requestLength = 0;
+                var messages = new List<string>();
+                for (var i = 1; i < 100; i++)
                 {
-                    Thread.Sleep(100);
+                    var message = i + Environment.NewLine;
+                    requestLength += message.Length;
+                    messages.Add(message);
                 }
 
-                var stringBuilder = new StringBuilder();
-                var buffer = new byte[4096];
-                int size;
-                while ((size = socket.Receive(buffer, buffer.Length, SocketFlags.None)) != 0)
+                await connection.Send(
+                    "POST /ReadAndWriteEchoLines HTTP/1.0",
+                    $"Content-Length: {requestLength}",
+                    "Host: localhost",
+                    "",
+                    "");
+
+                await connection.Receive(
+                    "HTTP/1.1 200 OK",
+                    "");
+                await connection.ReceiveHeaders();
+
+                foreach (var message in messages)
                 {
-                    stringBuilder.Append(Encoding.ASCII.GetString(buffer, 0, size));
+                    await connection.Send(message);
+                    await connection.Receive(message);
                 }
 
-                Assert.Contains(new StringBuilder().Insert(0, "hello world", 100).ToString(), stringBuilder.ToString());
+                await connection.Send("\r\n");
+                await connection.WaitForConnectionClose();
```

Comment:
This test doesn't assert anything.

Reply:
Yes it does, `Receive` takes an argument and asserts that it received what expected.